overseer/src/lib.rs

```diff
@@ -121,6 +121,35 @@ enum OverseerMessage<M: Debug, I> {
 	},
 }
 
+enum Event {
+	BlockImport,
+	BlockFinalized,
+	Stop,
+}
+
+/// A handler used to communicate with the `Overseer`.
+#[derive(Clone)]
+pub struct OverseerHandler {
+	events_tx: mpsc::Sender<Event>,
+}
+
+impl OverseerHandler {
+	/// Inform the `Overseer` that that some block was imported.
+	pub async fn block_imported(&mut self) -> SubsystemResult<()> {
```

Comment:
I'm in favor of the implication here that the overseer can be decoupled from a `BlockchainEvents` provider with this approach.

Reply:
Could you elaborate, I am not sure I quite understand, is this approach ok or should be changed?

Reply:
I mean that this structure of the code seems useful for building the `StartWork` and `StopWork` events generator in a clean way, no problem with it